File: RPI/Node.py
class Node:
    g=h=f=0
    nodeType = 0
    pos = [0,0]
    parent = 0
    obsDir = ''
    def __init__(self,x,y):
        self.pos = [x,y]
        
    def setObstacle(self):
        self.nodeType = 1
        
    def removeObstacle(self):
        self.nodeType = 0
    
    def printNode(self):
        print("Node Pos:["+str(self.pos[0])+","+str(self.pos[1])+"]" )
        print("G="+str(self.g)+"H="+str(self.h))
    
if __name__ == "__main__":
    n = Node(3,4)
    p = Node(2,5)
    n.parent = p
    #print(str(n.parent.pos[1]))

File: Algorithm/simulator/Node.java
package simulator;

import java.awt.*;
import java.util.ArrayList;
import java.util.Arrays;

public class Node{
	private float g,h,f;
	private int type;//type 0=available node,1=obstacle
	private int[] pos = new int[2];
	private Node parent;

	public Node(int x, int y){
		this.pos[0] = x;
		this.pos[1] = y;
		this.g=0;
		this.h=0;
		this.f=0;
		this.type = 0;
	}
	
	
	public void setObstacle(){ type = 1;}
	
	public void remObstacle(){ type = 0;}
	
	public int getType(){ return type; }
	
	public int[] getPos(){
		return pos;
	}
	
	public void setVal(float g, float h){
		this.g = g;
		this.h = h;
		this.f = g+h;
	}
	
	public float getG(){ return g;}
	public float getH(){ return h;}
	public float getF(){ return f;}
	
	public void setParent(Node p){parent = p;}
	public Node getParent(){return parent;}
	
	public void printNode(){
		System.out.println("Node pos: " + Arrays.toString(pos)+"\nG = "+g+"\nH = "+h+"\nF = "+f);
	}
	
}